up_client_zenoh/upclientzenoh.py

```diff
@@ -0,0 +1,463 @@
+from uprotocol.proto.ustatus_pb2 import UStatus, UCode
+from uprotocol.rpc.rpcclient import RpcClient
+from uprotocol.transport.builder.uattributesbuilder import UAttributesBuilder
+from uprotocol.transport.ulistener import UListener
+from uprotocol.transport.utransport import UTransport
+from uprotocol.transport.validate.uattributesvalidator import Validators
+from uprotocol.uri.factory.uresource_builder import UResourceBuilder
+from uprotocol.uri.validator.urivalidator import UriValidator
+from zenoh import open as zenoh_open, Subscriber, Session, Encoding, Config, Query, Sample, Value, Queryable
+
+from up_client_zenoh.zenohutils import ZenohUtils
+
+logger = logging.getLogger(__name__)
+
+
+class UPClientZenoh(UTransport, RpcClient):
+
+    def __init__(self, session: Session, subscriber_map: Dict[Tuple[str, UListener], Subscriber],
+                 queryable_map: Dict[Tuple[str, UListener], Queryable], query_map: Dict[str, Query],
+                 rpc_callback_map: Dict[str, UListener], uauthority: UAuthority, uentity: UEntity) -> None:
+        self.session = session
+        self.subscriber_map = subscriber_map
+        self.queryable_map = queryable_map
+        self.query_map = query_map
+        self.rpc_callback_map = rpc_callback_map
+        self.source_uuri = UUri(authority=uauthority, entity=uentity)
+        self.rpc_callback_lock = Lock()
+        self.queryable_lock = Lock()
+        self.subscriber_lock = Lock()
+
+    @classmethod
+    def new(cls, config: Config, uauthority: UAuthority, uentity: UEntity) -> 'UPClientZenoh':
+        session = zenoh_open(config)
+        return cls(session, {}, {}, {}, {}, uauthority, uentity)
+
+    def get_response_uuri(self) -> UUri:
+        new_source = self.source_uuri
+        new_source.resource.CopyFrom(UResourceBuilder.for_rpc_response())
+        return new_source
+
+    def send_publish_notification(self, zenoh_key: str, payload: UPayload, attributes: UAttributes) -> UStatus:
+        # Get the data from UPayload
+
+        if not payload.value:
+            msg = "The data in UPayload should be Data::Value"
+            print(f"ERROR: {msg}")
+            return UStatus(code=UCode.INVALID_ARGUMENT, message=msg)
+
+        buf = payload.value
+
+        # Transform UAttributes to user attachment in Zenoh
+        attachment = ZenohUtils.uattributes_to_attachment(attributes)
+        if not attachment:
+            msg = "Unable to transform UAttributes to attachment"
+            print(f"ERROR: {msg}")
+            return UStatus(code=UCode.INVALID_ARGUMENT, message=msg)
+
+        # Map the priority to Zenoh
+        priority = ZenohUtils.map_zenoh_priority(attributes.priority)
+        if not priority:
+            msg = "Unable to map to Zenoh priority"
+            print(f"ERROR: {msg}")
+            return UStatus(code=UCode.INVALID_ARGUMENT, message=msg)
+
+        try:
+            # Simulate sending data
+            print(f"Sending data to Zenoh with key: {zenoh_key}")
```

Comment:
Debug info? Maybe should not use print here.

Reply:
Done